Empire/Empire.cpp

```diff
@@ -3386,6 +3388,20 @@ void Empire::UpdateOwnedObjectCounters() {
         if (!ship->SpeciesName().empty())
             m_species_ships_owned[ship->SpeciesName()]++;
         m_ship_designs_owned[ship->DesignID()]++;
+        // parts
+        const ShipDesign* design = ship->Design();
+        if (!design)
+            continue;
+        const std::vector<std::string>& parts = design->Parts();
+        for (const std::string& part : parts) {
+            const PartType* part_type = GetPartType(part);
+            if (!part_type)
+                continue;
+            if (!part_type->Name().empty())
```

Comment:
should check for blank name before looking up, not after, as the latter will never return an empty name if the lookup worked.